bin/propolis-server/src/lib/server.rs

```diff
@@ -46,16 +47,22 @@ use crate::vnc::PropolisVncServer;
 pub struct MetricsEndpointConfig {
     /// The address at which the Oximeter endpoint will be hosted (i.e., this
     /// server's address).
-    pub propolis_addr: SocketAddr,
+    pub listen_addr: IpAddr,
 
     /// The address of the Nexus instance with which we should register our own
     /// server's address.
-    pub metric_addr: SocketAddr,
+    ///
+    /// If this is None _and the listen address is IPv6_, then internal DNS will
+    /// be used to register for metrics.
+    pub registration_addr: Option<SocketAddr>,
 }
 
 impl MetricsEndpointConfig {
-    pub fn new(propolis_addr: SocketAddr, metric_addr: SocketAddr) -> Self {
-        Self { propolis_addr, metric_addr }
+    pub fn new(
+        listen_addr: IpAddr,
+        registration_addr: Option<SocketAddr>,
+    ) -> Self {
+        Self { listen_addr, registration_addr }
     }
 }
```

Comment:
Since all the members are `pub`, why not do direct struct initialization where this is used?

Reply:
Done in 8a71e3d